src/components/product/OutOfStock/OutOfStock.tsx

```diff
@@ -24,48 +26,94 @@ export interface OutOfStockProps {
   title?: string
   /**
    * Notification icon.
-   * @default <Icon name="Bell" />
+   * @default <Icon name="BellRinging" />
    */
   icon?: ReactElement
+  /**
+   * Notification icon.
+   * @default <Icon name="BellRinging" />
+   */
+  notificationIcon?: ReactElement
   /**
    *
    */
   onSubmit: (value: string) => void
 }
 
 function OutOfStock(props: OutOfStockProps) {
+  const defaultButtonTxt = 'Notify me'
+  const defaultIcon = (
+    <Icon name="BellRinging" weight="bold" width={16} height={16} />
+  )
+
+  const [btTxt, setBtTxt] = useState(defaultButtonTxt)
+  const [disabled, setDisabled] = useState(false)
+  const [iconButton, setIconButton] = useState(defaultIcon)
+  const [email, setEmail] = useState('')
+
   const {
     title = 'Out of Stock',
     notificationMsg = 'Notify me when available',
-    buttonTxt = 'Send',
-    icon = <Icon name="Bell" />,
+    buttonTxt = btTxt,
+    icon = iconButton,
+    notificationIcon = defaultIcon,
     onSubmit,
     testId = 'store-out-of-stock',
   } = props
 
-  const [email, setEmail] = useState('')
+  const reset = () => {
+    setIconButton(defaultIcon)
+    setBtTxt(defaultButtonTxt)
+    setDisabled(false)
+
+    setEmail('')
+  }
+
+  const handleSubmit = async () => {
+    setDisabled(true)
+    setIconButton(<Icon name="Ellipsis" weight="bold" width={16} height={16} />)
+
+    try {
+      onSubmit(email)
+      setIconButton(
+        <Icon name="Checked" weight="bold" width={16} height={16} />
+      )
+      setBtTxt('Subscribed successfully')
+
+      // Return to original state after 2s
+      await new Promise((r) => setTimeout(r, 2000)).then(reset)
+    } catch (err) {
+      // TODO: Display error below Input component
+      console.error(err.Message)
+      reset()
+    }
+  }
 
   return (
     <div data-store-out-of-stock data-testid={testId} aria-live="polite">
-      <p>{title}</p>
-      <p>
-        {icon} {notificationMsg}
-      </p>
-      <Input
-        aria-label="Email"
-        value={email}
-        onChange={(e) => setEmail(e.target.value)}
-        placeholder="Email"
-      />
-      <Button
-        variant="primary"
-        icon={<Icon name="Bell" width={16} height={16} />}
-        iconPosition="left"
-        onClick={() => onSubmit(email)}
-      >
-        {buttonTxt}
-      </Button>
-      {/* TODO: Display success alert if onSubmit succeed */}
+      <div className="text__title-subsection">{title}</div>
+      <div>
+        {notificationIcon} {notificationMsg}
+      </div>
+      <div>
+        <Input
+          data-store-out-of-stock-input
+          aria-label="Email"
+          value={email}
+          onChange={(e) => setEmail(e.target.value)}
+          placeholder="Email"
+        />
+        <Button
+          disabled={disabled}
+          data-store-out-of-stock-button
+          variant="primary"
+          icon={icon}
+          iconPosition="left"
+          onClick={() => handleSubmit()}
```

Comment:
I think this should also work if don't need to pass any argument. Here we can pass the reference to the function to be called as well.
```suggestion
          onClick={handleSubmit}
```